---
title: "Gene profiling analyses"
author: "Jenna Crowe-Riddell"
date: "28 September 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Background
What we did to get the data:

* 12 samples were sequenced (Illumina RNA-seq) from three species of sea snake (*Aipysurus laevis, Aipysurus tenuis, Hydrophis major*) and two species of terrestrial snakes (*Notechis scutatus, Bitis rhinoceros*)
* The samples were taken from various organs including tail skin, body skin, retina, liver, heart, testis, vomeranasal
* RNA-seq reads were cleaned up (fast QC, trimmed adapters, etc.)
* Reads were aligned to reference genome (currently no genome for Aipysurus laevis, but close relative King cobra is available from [ncbi](https://www.ncbi.nlm.nih.gov/pubmed/24297900) 
* Reads were quantified using [RSEM pipeline](http://deweylab.biostat.wisc.edu/rsem/README.html#compilation), which generates expected counts, FPKM and TPM


## RNA read count data
Consists of table of expected counts for each 'gene id' (King cobra) and snake sample. Let's view the first 6 lines of count table:
```{r include = FALSE}
library(knitr)
library(magrittr)
library(tibble)
library(readr)
library(dplyr)
library(ggplot2)
library(limma)
library(edgeR)
library(RColorBrewer)
library(tidyverse)

```

```{r load dge.RData, include = FALSE}
load("dge.RData")
names(dge) <- gsub("_count","",names(dge))
```

```{r load sampleinfo.RData, include = FALSE}
load("sampleinfo.RData")
```

```{r}
head(dge)
```

Can also view our sample information:
```{r}
names(sampleinfo)
```

# Clustering data 

## MDS plot
The function plotMDS draws a multi-dimensional scaling plot of the RNA samples in which distances correspond to leading log-fold-changes between each pair of RNA samples. The leading log-fold-change is the average (root-mean-square) of the largest absolute log-foldchanges between each pair of samples. This plot can be viewed as a type of unsupervised clustering. 

To see how the RNA samples are clustering, we need to transform the count table (of expected read counts) into a data matrix and DGElist object, then normalise using the 'calcNormFactors()' function.
```{r, include=TRUE}
dge %<>% column_to_rownames("gene_id") %>%
  as.matrix() %>%
  edgeR::DGEList() %>%
  calcNormFactors()
```
Then plot the MDS using 'plotMDS' function and ggplot2 (to overlay sample information and make it look pretty).
```{r include= FALSE}
mds <- plotMDS(dge, gene.selection = "common", method = "logFC")
```

```{r echo= FALSE}
## Slightly ganky way of doing this, but extracting MDS coordinate object and joining with sample information.
plot <- mds@.Data[[3]] %>%
  as.data.frame() %>%
  rownames_to_column("group") %>%
  set_colnames(c("group","x","y")) %>%
  left_join(sampleinfo, by = c("group"="gene_id"))

## Plotting MDS using ggplot (nicer plot), includes overlap of experimental condition and species
plot %>%
  ggplot(aes(x, y, colour = experiment, shape = species, label = tissue)) +
  geom_point(size = 4) +
  geom_text(hjust = 0, nudge_x = 0.2, check_overlap = TRUE) + 
  theme_bw(base_size = 16) +
  theme(legend.text = element_text(size = 12)) + 
  labs(x = "Dimension 1",
       y=  "Dimension 2")
```









## Dendrogram
